5.0.0 Upgrade Considerations/ Tidy Software Upgrade Doc (#693)
astronomer/docs: software/manage-airflow-versions.md

> **Note:** In November of 2020, Astronomer migrated its Docker Registry from [Docker Hub](https://hub.docker.com/r/astronomerinc/ap-airflow) to [Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags) due to a [change](https://www.docker.com/blog/what-you-need-to-know-about-upcoming-docker-hub-rate-limiting/) in Docker Hub's rate limit policy. If you're using a legacy `astronomerinc/ap-airflow` image, replace it with a corresponding `quay.io/astronomer/ap-airflow` image to avoid rate limiting errors from DockerHub when you deploy to Astronomer (e.g. `toomanyrequests: You have reached your pull rate limit`).

> ### 3. Test Your Upgrade (_Optional_)

Reviewer: ```suggestion
### 3. Test Your Upgrade Locally (_Optional_)
```

> ### Step 3: Test Your Upgrade (_Optional_)

Reviewer 2: ```suggestion
## Step 3: Test Your Upgrade (_Optional_)
```

I think the "steps" are 2 deep, no? Otherwise we should remove "Step".
```suggestion
### 3. Test Your Upgrade (_Optional_)
```

> To test your upgrade on your local machine, make sure to save your changes and run the following from your command line:

Reviewer: ```suggestion
To test a new version of Astronomer Certified on your local machine, save all changes to your `Dockerfile` and run:
```

```sh
$ astro dev stop
```

> This will stop all 3 running Docker containers for each of the necessary Airflow components (Webserver, Scheduler, Postgres). From here, you can apply your changes and deploy them locally by running the following:

Reviewer: ```suggestion
This will stop all 3 running Docker containers for each of the necessary Airflow components (Webserver, Scheduler, Postgres). Then, apply your changes locally by running:
```

```sh
$ astro dev start
```

### 4. Deploy to Astronomer

> To push your upgrade to your Deployment, run:

Reviewer: ```suggestion
To push your upgrade to a Deployment on Astronomer Software, run:
```

```sh
astro deploy
```

:::caution

> Upgrading a Software Deployment to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than normal, with some upgrades taking anywhere from 10 minutes to an hour. If you need to minimize the upgrade time for a given Deployment, reach out to [Astronomer Support](https://support.astornomer.io) and request for Astronomer to remove unnecessary records from your metadata DB.

Reviewer: ```suggestion
[Apache Airflow 2.3.0](https://airflow.apache.org/docs/apache-airflow/2.3.0/release_notes.html#airflow-2-3-0-2022-04-30) requires a schema change to the Airflow metadata database. This means that upgrading a Deployment on Astronomer to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than usual. Depending on the size of your metadata database, upgrades can take anywhere from 10 minutes to an hour. During this time, scheduled tasks will continue to execute but new tasks will not be scheduled.

If you need to minimize the upgrade time of a given Deployment, reach out to [Astronomer Support](https://support.astornomer.io). On request, Astronomer can remove some historical records from your metadata database to lower the time it takes for your database to apply schema changes.
```

:::

> ### 5. Confirm your version in the Airflow UI

Reviewer 2: ```suggestion
## Step 3. Confirm your version in the Airflow UI
```

I think this should be up 1 level, making it 2 deep (also making the 2 headers below right before the change).

> ### Step 4: Confirm your version in the Airflow UI

Reviewer 2: ```suggestion
### Step 5: Confirm your version in the Airflow UI
```

Reviewer 2: Actually, I think this should be step 3 instead.

```suggestion
### Step 3: Confirm your version in the Airflow UI
```

And steps 3 and 4 above should just be "3." and "4.".

> **Note:** In November of 2020, Astronomer migrated its Docker Registry from [Docker Hub](https://hub.docker.com/r/astronomerinc/ap-airflow) to [Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags) due to a [change](https://www.docker.com/blog/what-you-need-to-know-about-upcoming-docker-hub-rate-limiting/) in Docker Hub's rate limit policy. If you're using a legacy `astronomerinc/ap-airflow` image, replace it with a corresponding `quay.io/astronomer/ap-airflow` image to avoid rate limiting errors from DockerHub when you deploy to Astronomer (e.g. `toomanyrequests: You have reached your pull rate limit`).

### 3. Test Your Upgrade Locally (_Optional_)

To test a new version of Astronomer Certified on your local machine, save all changes to your `Dockerfile` and run:

```sh
$ astro dev stop
```

This will stop all 3 running Docker containers for each of the necessary Airflow components (Webserver, Scheduler, Postgres). Then, apply your changes locally by running:

```sh
$ astro dev start
```

### 4. Deploy to Astronomer

To push your upgrade to a Deployment on Astronomer Software, run:

```sh
astro deploy
```

:::caution

> Due to a schema change in the Airflow metadata database, upgrading a Software Deployment to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than usual. Depending on the size of your metadata database, upgrades can take anywhere from 10 minutes to an hour or longer depending on the number of task instances that have been recorded in the Airflow metadata database. During this time, scheduled tasks will continue to execute but new tasks will not be scheduled.

Reviewer: ```suggestion
Due to a schema change in the Airflow metadata database, upgrading a Deployment on Astronomer to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than usual. Depending on the number of task instances that have been recorded in the Airflow metadata database, upgrades can take anywhere from 10 minutes to one hour or longer. During this time, scheduled tasks will continue to execute but new tasks will not be scheduled.
```

The two "depending on" was redundant!

> If you need to minimize the upgrade time for a given Deployment, reach out to [Astronomer Support](https://support.astornomer.io).

Author: ```suggestion
If you need to minimize the upgrade time for a given Deployment, reach out to [Astronomer Support](https://support.astronomer.io). Note that
```

> **Note:** In November of 2020, Astronomer migrated its Docker Registry from [Docker Hub](https://hub.docker.com/r/astronomerinc/ap-airflow) to [Quay.io](https://quay.io/repository/astronomer/ap-airflow?tab=tags) due to a [change](https://www.docker.com/blog/what-you-need-to-know-about-upcoming-docker-hub-rate-limiting/) in Docker Hub's rate limit policy. If you're using a legacy `astronomerinc/ap-airflow` image, replace it with a corresponding `quay.io/astronomer/ap-airflow` image to avoid rate limiting errors from DockerHub when you deploy to Astronomer (e.g. `toomanyrequests: You have reached your pull rate limit`).

### 3. Test Your Upgrade Locally (_Optional_)

To test a new version of Astronomer Certified on your local machine, save all changes to your `Dockerfile` and run:

```sh
$ astro dev stop
```

This will stop all 3 running Docker containers for each of the necessary Airflow components (Webserver, Scheduler, Postgres). Then, apply your changes locally by running:

```sh
$ astro dev start
```

### 4. Deploy to Astronomer

To push your upgrade to a Deployment on Astronomer Software, run:

```sh
astro deploy
```

:::caution

> Due to a schema change in the Airflow metadata database, upgrading a Software Deployment to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than usual. Depending on the size of your metadata database, upgrades can take anywhere from 10 minutes to an hour. During this time, scheduled tasks will continue to execute but new tasks will not be scheduled.

Reviewer 3: ```suggestion
Due to a schema change in the Airflow metadata database, upgrading a Software Deployment to [AC 2.3.0](https://github.com/astronomer/ap-airflow/blob/master/2.3.0/CHANGELOG.md) can take significantly longer than usual. Depending on the size of your metadata database, upgrades can take anywhere from 10 minutes to an hour or longer depending on the number of task instances that have been recorded in the Airflow metadata database. During this time, scheduled tasks will continue to execute but new tasks will not be scheduled.
```

> If you need to minimize the upgrade time for a given Deployment, reach out to [Astronomer Support](https://support.astornomer.io) and request for Astronomer to remove unnecessary records from your metadata DB.

Reviewer 3: I feel like this sentence doesn't really convey that we'll need to have a conversation about what is "unnecessary" to them. That depends on their business requirements.

Author: I'll just remove the "how" bit of this sentence, since it's not really important to know at this stage. Astronomer will then presumably walk through how to minimize the upgrade time based on their systems, so there's no real need for the user to walk in knowing exactly what needs to be done in this case.
